packages/bookshop/srv/admin-service.js

```diff
@@ -1,106 +1,99 @@
-        ShippingAddresses,
-        ownAddresses,
-        remoteAddresses
-      )
-      await tx.run(updateQueries)
-    } catch (e) {
-      console.error(e)
-    }
-  }
-})
 
-async function _readAddresses (req) {
-  console.log('Addresses', ShippingAddresses)
-  const BusinessPartner = req.user.id
-  const txExt = bupaSrv.transaction(req)
-  const selectQuery = req.query
-    .from(ShippingAddresses)
-    .where({ BusinessPartner })
-
-  try {
-    return txExt.run(selectQuery)
-  } catch (e) {
-    console.log(e)
-  }
-}
+// We are mashing up three services...
+const admin = cds.connect.to ('AdminService')
+const bupa = cds.connect.to('API_BUSINESS_PARTNER')
+const db = cds.connect.to('db')
 
-async function _fillAddress (req) {
-  if (req.data.shippingAddress_AddressID) {
-    const BusinessPartner = req.user.id
-    const txExt = bupaSrv.transaction(req)
-    try {
-      const response = await txExt.run(
-        SELECT.from(ShippingAddresses).where({
-          AddressID: req.data.shippingAddress_AddressID,
-          BusinessPartner
-        })
-      )
-      if (response && response.length === 1) {
-        const tx = cds.transaction(req)
-        const insertQuery = INSERT.into(ShippingAddresses).entries(response)
-        await tx.run(insertQuery)
-      }
-    } catch (e) {}
-  }
-}
+// Reflected entities for local database
+const { Books, Addresses } = db.entities
 
-async function _reduceStock (req) {
-  const { Items: OrderItems } = req.data
-  if (OrderItems && OrderItems.length > 0) {
-    const all = await cds.transaction(req).run(
-      OrderItems.map(order =>
-        UPDATE(Books)
-          .set('stock -=', order.amount)
-          .where('ID =', order.book_ID)
-          .and('stock >=', order.amount)
-      )
-    )
-    all.forEach((affectedRows, i) => {
-      if (affectedRows === 0)
-        req.error(
-          409,
-          `${OrderItems[i].amount} exceeds stock for book #${
-            OrderItems[i].book_ID
-          }`
-        )
+// Fetch current user's addresses from S/4 for ValueHelp.
+module.exports = (admin => {
+  admin.on ('READ', 'usersAddresses', async (req) => {
+    // const UsersAddresses = req.query.from (Addresses) .where ({ BusinessPartner: req.user.id })
+    // FIXME: Again that absolutely useless error message:
+    // [2019-12-16T20:30:14.106Z | ERROR | 1940862]: The server does not support the functionality required to fulfill the request
+    // FIXME: Even worse: click Orders Edit ->
+    // [2019-12-16T20:38:52.918Z | WARNING | 1575675]: Not Found
+    const { A_BusinessPartnerAddress:Addresses } = bupa.entities
+    const UsersAddresses = SELECT.from (Addresses, a => {
+      a.AddressID.as('ID'),
+      a.BusinessPartner,
+      a.Country.as('country'),
+      a.CityName.as('cityName'),
+      a.PostalCode.as('postalCode'),
+      a.StreetName.as('streetName'),
+      a.HouseNumber.as('houseNumber')
+    }) .where ({ BusinessPartner: req.user.id })
```

Comment:
I think I know why automatic aliasing doesn't work here: You defined the entity on the external service itself. I have a check that I will only 'follow the projection', if the entity is _not_ part of the transaction's service and the target of the projection _is_ part of the transaction's service. Here:`Addresses` _is_ part of the transaction's service.

Reply:
Better check: If the transaction's service is of kind odata/rest: _Always_ follow projections.

Reply:
But there's another problem:
```javascript
bupa.entities
```
does _not_ have the entity `Addresses` if `bupa` is of kind REST/OData and _not_ mocked.

Reply:
So apparently external service extension are not taken into account for non-mocked REST/OData services.